action/candidate_register.go

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) 2020 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package action
+
+import (
+	"math/big"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/iotex-address/address"
+	"github.com/iotexproject/iotex-core/pkg/util/byteutil"
+	"github.com/iotexproject/iotex-core/pkg/version"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+)
+
+const (
+	// CandidateRegisterPayloadGas represents the CandidateRegister payload gas per uint
+	CandidateRegisterPayloadGas = uint64(100)
+	// CandidateRegisterBaseIntrinsicGas represents the base intrinsic gas for CandidateRegister
+	CandidateRegisterBaseIntrinsicGas = uint64(10000)
+)
+
+// CandidateRegister is the action to register a candidate
+type CandidateRegister struct {
+	AbstractAction
+
+	name            string
```

Comment:
suggest to use CandName here
then we can use it to directly access the delegate map